### PR TITLE
Convert AnimationBone and Animator to unique ptr

### PR DESCRIPTION
**Pull request by ShFil119**

### CODE REVIEW

**Comment by ShFil119** [comment]
Worth noted is that - after this pull it will be easy to write code which clears resources after cutscenes.

File: rwengine/src/ai/CharacterController.cpp [hunk 1]
```diff
@@ -669,7 +669,7 @@ bool Activities::UseItem::update(CharacterObject *character,
     auto world = character->engine;
     const auto &weapon = world->data->weaponData.at(itemslot);
     auto &state = character->getCurrentState().weapons[itemslot];
-    auto animator = character->animator;
+    auto& animator = character->animator;
```
Comment (Lihis) [inline rwengine/src/ai/CharacterController.cpp]: Inconsistent placement of `&`.

File: rwengine/src/objects/CharacterObject.cpp [hunk 2]
```diff
@@ -316,12 +316,12 @@ void CharacterObject::changeCharacterModel(const std::string& name) {
     auto newmodel = engine->data->loadClump(modelName + ".dff");
 
     if (animator) {
```
Comment (danhedron) [inline rwengine/src/objects/CharacterObject.cpp]: The if can be removed and just call `animator.reset()` instead.

**Comment by ShFil119** [comment]
Updated. :)

**Pull request merged**